src/Native/JsArray.js

```diff
@@ -0,0 +1,84 @@
+// import Maybe
+
+var _elm_lang$core$Native_JsArray = function() {
+/* A thin, but still immutable, wrapper over native Javascript arrays. */
+
+// An empty array
+var empty = {
+    ctor: 'JsArray',
+    _0: []
+};
+
+function singleton(val) {
+    return {
+        ctor: 'JsArray',
+        _0: [val]
+    };
+}
+
+function length(arr) {
+    return arr._0.length;
+}
+
+function get(idx, arr) {
+    var val = arr._0[idx];
+    if (val) {
+        return _elm_lang$core$Maybe$Just(val);
+    }
+
+    return _elm_lang$core$Maybe$Nothing;
+}
+
+function set(idx, val, arr) {
+    var copy = arr._0.slice();
```

Comment:
There are no bounds checking here. From the comment, 
```
Sets an element at a given index. Returns an unmodified array if
 +index is out of bounds.
```

Check the array size, and make sure `0 <= idx < array.length`